File: Sources/MicrosoftOrganizationChartReader/Records/ConnectorRecord.swift
//
//  ConnectorRecord.swift
//  
//
//  Created by Hugh Bellamy on 27/01/2021.
//

import DataStream

public struct ConnectorRecord {
    public let size: UInt16
    public let unknown1: UInt16
    public let unknown2: UInt16
    public let unknown3: UInt16
    public let targetIndex: UInt16
    public let targetLevel: UInt16
    public let unknown4: UInt16
    public let unknown5: UInt16
    public let unknown6: UInt16
    public let unknown7: UInt16
    public let unknown8: UInt16
    public let unknown9: UInt16
    public let unknown10: UInt16

    public init(dataStream: inout DataStream) throws {
        self.size = try dataStream.read(endianess: .littleEndian)
        self.unknown1 = try dataStream.read(endianess: .littleEndian)
        self.unknown2 = try dataStream.read(endianess: .littleEndian)
        self.unknown3 = try dataStream.read(endianess: .littleEndian)
        self.targetIndex = try dataStream.read(endianess: .littleEndian)
        self.targetLevel = try dataStream.read(endianess: .littleEndian)
        self.unknown4 = try dataStream.read(endianess: .littleEndian)
        self.unknown5 = try dataStream.read(endianess: .littleEndian)
        self.unknown6 = try dataStream.read(endianess: .littleEndian)
        self.unknown7 = try dataStream.read(endianess: .littleEndian)
        self.unknown8 = try dataStream.read(endianess: .littleEndian)
        self.unknown9 = try dataStream.read(endianess: .littleEndian)
        self.unknown10 = try dataStream.read(endianess: .littleEndian)
    }
}

File: Explorer/MicrosoftOrganizationChartReader/Records/ConnectorRecord.cs
using System.ComponentModel;
using System.IO;

namespace OrganizationChartReader
{
    public class ConnectorRecord
    {
        public ushort Size { get; set; }

        public ushort OriginIndex { get; set; }

        public ushort OriginLevel { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown3 { get; set; }

        public ushort TargetIndex { get; set; }

        public ushort TargetLevel { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown4 { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown5 { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown6 { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown7 { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown8 { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown9 { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown10 { get; set; }

        public ConnectorRecord(BinaryReader reader)
        {
            Size = reader.ReadUInt16();
            OriginIndex = reader.ReadUInt16();
            OriginLevel = reader.ReadUInt16();
            Unknown3 = reader.ReadUInt16();
            TargetIndex = reader.ReadUInt16();
            TargetLevel = reader.ReadUInt16();
            Unknown4 = reader.ReadUInt16();
            Unknown5 = reader.ReadUInt16();
            Unknown6 = reader.ReadUInt16();
            Unknown7 = reader.ReadUInt16();
            Unknown8 = reader.ReadUInt16();
            Unknown9 = reader.ReadUInt16();
            Unknown10 = reader.ReadUInt16();
        }
    }
}
